diff --git a/dist/index.js b/dist/index.js
index b17b126740f87c6ac36ce70c841be3c773150411..ad4753913b9d4c1b09f6dd6a47d87e1b991e40bc 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -74,15 +74,16 @@ var json = /* @__PURE__ */ __name((url, flags, options) => {
   const process2 = (0, import_node_child_process.spawn)(YTDLP_PATH, args(url, flags), options);
   return new Promise((resolve, reject) => {
     let output = "";
+    let errorOutput = "";
     process2.stdout?.on("data", (chunk) => {
       output += chunk;
     });
     process2.stderr?.on("data", (chunk) => {
-      output += chunk;
+      errorOutput += chunk;
     });
     process2.on("close", (code) => {
       if (code === 0) resolve(JSON.parse(output));
-      else reject(new Error(output));
+      else reject(new Error(errorOutput || output));
     });
     process2.on("error", reject);
   });
diff --git a/dist/index.mjs b/dist/index.mjs
index bf0733715e9229249d227c77a6d394084f879237..d2ca3a012a5556c1eed01f41c0e01f1cf0a6539c 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -45,15 +45,16 @@ var json = /* @__PURE__ */ __name((url, flags, options) => {
   const process2 = spawn(YTDLP_PATH, args(url, flags), options);
   return new Promise((resolve, reject) => {
     let output = "";
+    let errorOutput = "";
     process2.stdout?.on("data", (chunk) => {
       output += chunk;
     });
     process2.stderr?.on("data", (chunk) => {
-      output += chunk;
+      errorOutput += chunk;
     });
     process2.on("close", (code) => {
       if (code === 0) resolve(JSON.parse(output));
-      else reject(new Error(output));
+      else reject(new Error(errorOutput || output));
     });
     process2.on("error", reject);
   });
